infra/feast-operator/test/e2e/e2e_test.go

```diff
@@ -127,6 +127,117 @@ var _ = Describe("controller", Ordered, func() {
 			}
 			EventuallyWithOffset(1, verifyControllerUp, time.Minute, time.Second).Should(Succeed())
 
+			By("deploying the Simple Feast Custom Resource to Kubernetes")
+			cmd = exec.Command("kubectl", "apply", "-f",
+				"test/testdata/feast_integration_test_crs/v1alpha1_default_featurestore.yaml")
+			_, cmdOutputerr := utils.Run(cmd)
+			ExpectWithOffset(1, cmdOutputerr).NotTo(HaveOccurred())
+
+			By("validate the feast registry pod is up and running state.")
+			verifyFeastControllerUp := func() error {
+				// Get pod name
+
+				cmd := exec.Command("kubectl", "get",
+					"pods", "-l", "feast.dev/service-type=registry",
```

Comment:
i wonder if the better route might be to check the registry deployment object and wait for availability condition to be `true`. this will allow for easier testing around scaling later when we add that capability.

Reply:
I m checking the deployment status of registry, online and offline pods. Abstracted the code and it is much better now. PTAL and let me know if further changes are required.